webscraper/dataScraperAPI.js

```diff
@@ -10,65 +10,221 @@ function scrape() {
+                i +
+                ') > td:nth-child(3)',
+            )
+              .text()
+              .trim(),
+            windDirection: $(
+              'table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(' +
+                i +
+                ') > td:nth-child(4)',
+            )
+              .text()
+              .trim(),
+            windSpeed: $(
+              'table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(' +
+                i +
+                ') > td:nth-child(5)',
+            )
+              .text()
+              .trim(),
+          };
           j++;
         }
-      else 
         // do nothing
-        continue;
+        else continue;
       }
 
       // Scraping info for histSwellInfo
-      let histSwellTbodyLen = $("table.dataTable:nth-child(7) > tbody:nth-child(2)").children('tr').length;
+      let histSwellTbodyLen = $('table.dataTable:nth-child(7) > tbody:nth-child(2)').children(
+        'tr',
+      ).length;
       for (let i = 3; i < histSwellTbodyLen + 1; i++) {
-        histSwellInfo[i-3] = {
-          'month'            : $("table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(" + i + ") > td:nth-child(1)").text().trim(),
-          'day'              : $("table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(" + i + ") > td:nth-child(2)").text().trim(),
-          'time'             : $("table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(" + i + ") > td:nth-child(3)").text().trim(),
-          'waveHeight'       : $('table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(4)').text().trim(),
-          'swellHeight'      : $('table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(5)').text().trim(),
-          'swellPeriod'      : $('table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(6)').text().trim(),
-          'swellDirection'   : $('table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(7)').text().trim(),
-          'windWaveHeight'   : $('table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(8)').text().trim(),
-          'windWavePeriod'   : $('table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(9)').text().trim(),
-          'windWaveDirection': $('table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(10)').text().trim(),
-          'steepness'        : $('table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(11)').text().trim(),
-          'averageWavePeriod': $('table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(12)').text().trim(),
-        }
+        histSwellInfo[i - 3] = {
+          month: $(
+            'table.dataTable:nth-child(7) > tbody:nth-child(2) > tr:nth-child(' +
```

Comment:
> Took me a while to appreciate that code is for humans rather than machines. Keeping that perspective in mind helped me a lot.

Review your code with this principle in mind. The highlighted line is just an example of what could use a refactor. Due to syntax and styling of the document it takes me great effort to figure out what it's supposed to represent.

Additionally, consider implementing the refactor we previously discussed where you parse the text file (e.g., https://www.ndbc.noaa.gov/data/realtime2/44065.txt) instead of the HTML rich page that you are currently using. Not only will this speed up your API, but it will also reduce the complexity of your scraping process.